src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs

```diff
@@ -128,11 +129,18 @@ public override void OnStopActivity(Activity activity, object payload)
                     return;
                 }
 
-                var response = context.Response;
-                activity.AddTag(SemanticConventions.AttributeHttpStatusCode, response.StatusCode.ToString());
+                if (this.TryGetGrpcMethod(activity, out var grpcMethod))

```

Comment:
It looks like different tags are added depending on whether the GrpcMethod can be retrieved, is that correct?

For example:
DisplayName (currently commented out, AttributeRpcSystem, service, method, etc
vs
AttributeHttpStatusCode, Status

Should we drop the else so the later tags are always added?

Reply:
My decision here was intentional, though I did raise the question earlier whether all the `http.*` attributes should be omitted from the span in the context of a gRPC invocation.

Two things occur in this if/else:
1. The `http.status_code` attribute is set
2. The Status of the span gets set

At a minimum the else clause here should not set the span's status based on the HTTP status because the gRPC status should be used instead. Though, I could see an argument for leaving the `http.status_code` attribute. Other `http.*` attributes are added in `OnStartActivity`. Given that this PR currently leaves those untouched, then it probably makes sense to leave `http.status_code`.

I'm ok coming back to this question of keeping vs. omitting the `http.*` attributes and iterating on this instrumentation late, though I think I need to make sure the test actually reflects these decisions.

Reply:
I expanded the test to reflect this conversation.

https://github.com/open-telemetry/opentelemetry-dotnet/blob/8232d33812783bf1b89fd10b712b26681f052531/test/OpenTelemetry.Instrumentation.Grpc.Tests/GrpcTests.server.cs#L63-L72

Reply:
I think we discussed a long time ago, but I personally like having both the http and grpc tags in there. GRPC is not limited to working over HTTP, even though not many alternatives exist. I may be considered unnecessary details for now though 🤷 
@cijothomas @CodeBlanch What are your thoughts?

Reply:
> personally like having both the http and grpc tags in there

I agree that it probably makes sense to leave them in. I'm not necessarily advocating for one way or another at this point, but just to highlight some things for ongoing consideration:

* For the client side grpc-dotnet instrumentation we [discussed](https://github.com/open-telemetry/opentelemetry-dotnet/pull/687#discussion_r427659307) the possibility of allowing the suppression of the underlying HTTP span that is generated. There exist other [examples](https://github.com/open-telemetry/opentelemetry-dotnet/issues/893) where this is desired as well.
* For the ASP.NET Core server side instrumentation, things are a bit different, there is no "other" span. Attributes are applied to the span (activity) that ASP.NET Core started. The library itself adds attributes [here](https://github.com/grpc/grpc-dotnet/blob/9c294e0394388a96c17dd8b38ae0fbf5961c968c/src/Grpc.AspNetCore.Server/Internal/HttpContextServerCallContext.cs#L279-L283).

> GRPC is not limited to working over HTTP

Yes, true, and I definitely see value in reflecting transport type reflected on calls both server-side and client-side. This may be a thing that needs to be continued to be pushed at the spec level. I personally have experience with instrumenting WCF - which is a complex beast, and I've confronted a lot of these same considerations with it.

Slight tangent, but with WCF and perhaps eventually with gRPC, transport type used affects whether we can do auto-context propagation. For example, if the transport is a raw TCP channel, then there exist no structured headers mechanism to inject context - it has to be done manually.